packages/commonwealth/server/routes/status.ts

```diff
@@ -102,156 +102,52 @@ export const getUserStatus = async (models: DB, user: UserInstance) => {
-   */
-  const communityActivity = await getCommunityActivity(addresses);
-  const unseenPosts = {};
-  let query = ``;
-  let replacements: string[] = [];
-
-  // this loops through the communities for which we want to see if there are any new updates
-  // for each community a UNION SELECT query is appended to the query so that that communities updated threads are
-  // included in the final result. This method allows us to submit a single query for all the communities rather
-  // than a new query for each community
-  for (let i = 0; i < communityActivity.length; i++) {
-    const name = communityActivity[i][0];
-    const date = communityActivity[i][1];
-
-    if (!date) {
-      unseenPosts[name] = {};
-      continue;
-    }
-
-    // adds a union between SELECT queries if the number of SELECT queries is greater than 1
-    if (query !== '') query += ' UNION ';
-    // add the community and timestamp to replacements so that we can safely populate the query with dynamic parameters
-    replacements.push(name, date);
-    // append the SELECT query
-    query += `SELECT id, community_id
-              FROM "Threads"
-              WHERE community_id = ?
-                AND created_at > ?
-                AND deleted_at IS NULL`;
-    if (i === communityActivity.length - 1) query += ';';
-  }
-
-  // populate the query replacements and execute the query
-  const threadNumPromise = sequelize.query<
-    Pick<ThreadAttributes, 'id' | 'community_id'>
-  >(query, {
-    raw: true,
-    type: QueryTypes.SELECT,
-    replacements,
-  });
-
-  // wait for all the promises to resolve
-  const [starredCommunities, threadNum] = await Promise.all([
-    starredCommunitiesPromise,
-    threadNumPromise,
-  ]);
-
-  // this section iterates through the retrieved threads
-  // counting the number of threads and keeping a set of activePosts
-  // the set of activePosts is used to compare with the comments
-  // under threads so that there are no duplicate active threads counted
-  for (const thread of threadNum) {
-    if (!unseenPosts[thread.community_id])
-      unseenPosts[thread.community_id] = {};
-    unseenPosts[thread.community_id].activePosts
-      ? unseenPosts[thread.community_id].activePosts.add(thread.id)
-      : (unseenPosts[thread.community_id].activePosts = new Set([thread.id]));
-    unseenPosts[thread.community_id].threads
-      ? unseenPosts[thread.community_id].threads++
-      : (unseenPosts[thread.community_id].threads = 1);
-  }
-
-  // reset var
-  query = ``;
-  replacements = [];
-
-  // same principal as the loop above but for comments instead of threads
-  for (let i = 0; i < communityActivity.length; i++) {
-    const name = communityActivity[i][0];
-    const date = communityActivity[i][1];
-
-    if (!date) {
-      unseenPosts[name] = {};
+  // get ids of user-joined coommunities which have new content after user's last_active date per addresses
+  const joinedCommunityIdsWithNewContent: string[] = [];
```

Comment:
We should avoid loops with internal queries when possible. Looks like the intent of the query is to find recent content (threads or comments after a given date) for a given number of addresses linked to communities. 

You can try to put this query behind the new framework and optimize the execution plan... see examples in `/libs/mode/src/community/GetMembers.querry.ts` and `/libs/model/src/feed/GetUserActivity.query.ts`.

Following the new convention, something like `/libs/model/src/feed/GetNewContent.query.ts` ... you can name the file/query in a way that clearly describes the intent. This way we can expose it as a public GET endpoint (API) with tRPC.

In case we don't need this to be public, you can still encapsulate the query behind a function following a repository pattern. Let me know if you need help with this and we can assist.

Reply:
+1 - we should encapsulate this logic in a new query and write some load tests for it as I suspect it could grow increasingly expensive as the number of communities the user has joined grows.

Also, from what I've seen from @jessmart1213 designs for the sidebar we will eventually have counts for each community indicating new content so having a separate query should make this update easier in the future
e.g.
![Screenshot from 2024-07-29 16-34-53](https://github.com/user-attachments/assets/819308b9-28d9-4686-9367-482f4a7c240f)

Reply:
Not sure if i did [GetNewContent.query.ts](https://github.com/hicommonwealth/commonwealth/pull/8628/files#diff-bdbdbc1f810ed24c8f8e390f8ba3eef3d454f8fc374bc04386970b740162b8f1) correctly, plz lmk if any issues